### PR TITLE
EventManager Improvements

### PR DESCRIPTION
**Pull request by Pessimistress**
- Add `pointerleave` event (https://github.com/uber/deck.gl/issues/847)
- General code consistency
- Tests

### CODE REVIEW

File: src/controls/events/key-input.js [hunk 1]
```diff
@@ -1,71 +1,62 @@
...
+    this.enableUpEvent = this.options.enable;
+
+    this.events = KEY_EVENTS.concat(options.events || []);
 
     this.handleEvent = this.handleEvent.bind(this);
-    this.options.events.forEach(event => window.addEventListener(event, this.handleEvent));
-  }
 
-  destroy() {
-    this.options.events.forEach(event => window.removeEventListener(event, this.handleEvent));
```
Comment (ibgreen) [inline src/controls/events/key-input.js]: Cool, I'll test this on my branch and see if it still works.

File: src/controls/events/move-input.js [hunk 2]
```diff
@@ -16,65 +17,79 @@ export default class MoveInput {
...
 
   /**
    * Enable this input (begin processing events)
    * if the specified event type is among those handled by this input.
    */
   toggleIfEventSupported(eventType, enabled) {
-    if (EVENT_TYPE === eventType) {
-      this.options.enable = enabled;
+    if (eventType === MOVE_EVENT_TYPE) {
+      this.enableMoveEvent = enabled;
```
Comment (ibgreen) [inline src/controls/events/move-input.js]: This makes more sense, I couldn't see how the old system could be made to fit...

File: src/controls/generic-controls.js [hunk 3]
```diff
@@ -289,25 +289,31 @@ export default class GenericControls {
     }
 
     const KEY_BINDINGS = {
-      KeyW: 'moveForward',
+      w: 'moveForward',
+      W: 'moveForward',
       ArrowUp: 'moveForward',
```
Comment (ibgreen) [inline src/controls/generic-controls.js]: Don't the arrow keys need to be replaced with numbers or similar as well? 

Good if we can keep them working, they are really important to the controller use case...

File: src/controls/events/move-input.js [hunk 4]
```diff
@@ -16,65 +17,79 @@ export default class MoveInput {
...
-  set(options) {
-    Object.assign(this.options, options);
+    this.events.forEach(event => this.element.removeEventListener(event, this.handleEvent));
   }
 
   /**
    * Enable this input (begin processing events)
    * if the specified event type is among those handled by this input.
    */
   toggleIfEventSupported(eventType, enabled) {
```
Comment (ibgreen) [inline src/controls/events/move-input.js]: For API Audit: Something about the name of this method bothers me. Wonder why we didn't just call it `enableEvent` or `enableEventType`? 

It doesn't even seem to toggle, as it takes an enabled flag...

File: src/controls/generic-controls.js (hunk 3 above)
Comment (Pessimistress) [inline src/controls/generic-controls.js]: `event.key` produces slightly different values from `event.code` but is better supported cross browsers.

**Pull request merged**